### PR TITLE
[4393] Receptacles datasource is not filtering options selected in configuration

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4393

### CODE REVIEW

**Review by rart — CHANGES_REQUESTED** [review]
https://github.com/craftercms/craftercms/issues/4393#issuecomment-766803322

File: static-assets/components/cstudio-admin/mods/content-type-propsheet/boolean.js [hunk 1]
```diff
@@ -44,11 +44,23 @@ YAHOO.extend(
           _self.value = _self.getValue();
           updateFn(event, _self);
           CStudioAdminConsole.Tool.ContentTypes.visualization.render();
+          valueEl.dispatchEvent(new Event('propertyUpdate', { event }));
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/content-type-propsheet/boolean.js:48]: How about just subscribing to the change event of the control?

File: static-assets/components/cstudio-admin/mods/content-types.js [hunk 2]
```diff
@@ -2827,6 +2827,58 @@
...
+            });
+        } else {
+          $(element)
+            .parent()
+            .popover('destroy');
+        }
+      },
+
+      // TODO: handle validations with other types
+      isDependencyMet: function(dependency) {
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/content-types.js]: Add console logs for warning not supported stuff. Ping me to discuss message

**Review by rart — CHANGES_REQUESTED** [review]
Is this ready?

File: ui/app/src/utils/i18n-legacy.ts [hunk 3]
```diff
@@ -413,6 +413,15 @@ export const contentTypesMessages = defineMessages({
...
+  },
+  dependsOn: {
+    id: 'contentType.dependsOn',
+    defaultMessage: 'This property depends on "{dependency}"'
+  },
+  dependencyNotImplemented: {
+    id: 'contentType.dependencyNotImplemented',
+    defaultMessage:
+      '[{property}.js] Control dependency not implemented for input of type "{property}". Dependency will be ignored.'
   }
```
Comment (rart) [inline ui/app/src/utils/i18n-legacy.ts]: This was really just meant to be a console.log. No need for translation or UI popup etc. The file name should be fixed to the file name that issues the log so when we see it we can easily find, inspect and address if needed.

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Comment by sumerjabri** [comment]
@rart anything to be done regarding: https://app.codacy.com/gh/craftercms/studio-ui/pullRequest?prid=6840088

**Comment by rart** [comment]
> @rart anything to be done regarding: https://app.codacy.com/gh/craftercms/studio-ui/pullRequest?prid=6840088

no, @sumerjabri

**Pull request merged**